cmd/soroban-cli/src/commands/tx/new/create_claimable_balance.rs

```diff
@@ -0,0 +1,377 @@
+use clap::{command, Parser};
+use serde_json;
+use std::str::FromStr;
+
+use crate::{commands::tx, config::address, tx::builder, xdr};
+
+#[derive(Debug, serde::Deserialize, serde::Serialize, PartialEq)]
+#[serde(rename_all = "snake_case", tag = "type", content = "value")]
+pub enum PredicateJson {
+    Unconditional,
+    BeforeAbsoluteTime(String),
+    BeforeRelativeTime(u64),
+    Not(Box<PredicateJson>),
+    And(Vec<PredicateJson>),
+    Or(Vec<PredicateJson>),
+}
```

Comment:
Is it possible to use XDR-JSON for this, so that the json specified uses those types? Or does that not work which led to the type here?

Reply:
Good question. I didn’t even think of that. I’ll take a look at it tomorrow.

Reply:
@leighmcculloch great call, Leigh! It simplifies the whole thing a bunch.